### ISSUE TITLE
prop-types rule triggered when accessing second level properties using props[property]

### ISSUE THREAD

**Issue by BerkeleyTrue** [yannickcr/eslint-plugin-react#298]
The following will trigger prop-types rule. As is shown, I'm accessing a second level property of a props value which shouldn't trigger prop-types error.
```js
    handleSubmit(e) {
      e.preventDefault();
      const props = this.props;
      let valid = true;
      checkValidity.forEach((prop) => {
        // if value exist, check if it is valid
        if (props[prop].value && props[prop].type !== 'boolean') {
          valid = valid && !!props[prop].valid;
        }
      });
```

Error goes away if I change `props` to `foo` .

```js
    handleSubmit(e) {
      e.preventDefault();
      const foo = this.props;
      let valid = true;
      checkValidity.forEach((prop) => {
        // if value exist, check if it is valid
        if (foo[prop].value && foo[prop].type !== 'boolean') {
          valid = valid && !!foo[prop].valid;
        }
      });
```

Using
babel-eslint@4.1.4
eslint@1.8.0
eslint-plugin-react@3.7.1
